### ISSUE TITLE
error plotting less than 2 points using plot() method of Basemap

### ISSUE THREAD

**Issue by paitor** [matplotlib/basemap#213]
Hi 

Running python 2.7.8 and Basemap 1.0.7 (downloaded and installed today at about 1045 UTC) I fail to plot less than 3 points using the plot() method. I've tested this using the following script:
```python
#! /usr/bin/env python

import matplotlib
matplotlib.use('TkAgg')
import matplotlib.pyplot as plt
from mpl_toolkits.basemap import Basemap

bm = Basemap(llcrnrlon=0,llcrnrlat=-80,urcrnrlon=360,urcrnrlat=80,projection='mill')
fig=plt.figure(figsize=(8,4.5))   

# figure with three points work
bm.plot([10,15,20], [10,10,10], "ro", latlon=True)
plt.show()

# figure with less than three points fail
bm.plot([10,15], [10,10], "ro", latlon=True)
plt.show()
```
And the exception is:
```
:~> python TryBasemap.py 
Traceback (most recent call last):
  File "TryBasemap.py", line 16, in <module>
    bm.plot([10,15], [10,10], "ro", latlon=True)
  File "[...]/python2.7/site-packages/mpl_toolkits/basemap/__init__.py", line 536, in with_transform
    x = self.shiftdata(x)
  File "[...]/python2.7/site-packages/mpl_toolkits/basemap/__init__.py", line 4775, in shiftdata
    thresh = 360.-londiff_sort[-2]
IndexError: index -2 is out of bounds for axis 0 with size 1
```
Whilst the first plot is successful, the second fail

**Comment by guziy** [comment]
This is related to #191 ... Workaround is to not use latlon=True, but reproject the coordinates yourself. I am not sure if it'll help your next issue, but please try and report... 

I'll try to check if there is an non-intrusive way to change shiftdata so it works for 2 points...

Cheers

**Comment by paitor** [comment]
Well, my workaround for now is to simply duplicate the last point until I have three, seems to work.

**Comment by guziy** [comment]
)))

**Comment by guziy** [comment]
Let's see if maintainers notice this issue, and if they do not see any problems with your solution, you could implement it yourself and submit a pull request..

**Comment by paitor** [comment]
Well, that means I first have to figure out how to submit a pull request, but given that I lately have been whining on various github projects I really should get my lazy as around to do so, would certainly be more constructive

**Comment by WeatherGod** [comment]
The workaround is fine as a workaround, but not as a fix. The returned Collection would have extra elements that do not correspond to the original data. I am just going to need to roll up my sleeves and dig into that function.

**Comment by guziy** [comment]
I decided to pull this in order to not lose it)) If it is not in line with good practices it is OK to ignore it, or any request for improvement is always welcome)))
Cheers

**Comment by paitor** [comment]
Great, guizy, thanks :)

**Comment by WeatherGod** [comment]
closed via #215
